client/src/components/post/hooks/useLike.js

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { likePost } from '../api';
+
+const useLike = ({ likes, postId }) => {
+  const {
+    data: {
+      data: { _id: userId },
+    },
+  } = useQuery('fetchUser');
+
+  const userLikePostCheck = () => {
+    let check = false;
+    likes?.forEach((like) => {
+      if (like?.userId === userId) {
+        check = true;
+      }
+    });
+    return check;
+  };
+
+  const [checked, setChecked] = useState(userLikePostCheck());
```

Comment:
I think lazyloading the usestate function is better in this scenario. useState(() => userLikePostCheck());

Reply:
we take a look at it